src/Strategy/Python/Poetry/PyProject.hs

```diff
@@ -0,0 +1,311 @@
+  { sourcePath :: Text
+  }
+  deriving (Show, Eq, Ord)
+
+pyProjectPoetryPathDependencyCodec :: TomlCodec PyProjectPoetryPathDependency
+pyProjectPoetryPathDependencyCodec =
+  PyProjectPoetryPathDependency
+    <$> Toml.text "path" .= sourcePath
+
+newtype PyProjectPoetryUrlDependency = PyProjectPoetryUrlDependency
+  { sourceUrl :: Text
+  }
+  deriving (Show, Eq, Ord)
+
+pyProjectPoetryUrlDependencyCodec :: TomlCodec PyProjectPoetryUrlDependency
+pyProjectPoetryUrlDependencyCodec =
+  PyProjectPoetryUrlDependency
+    <$> Toml.text "url" .= sourceUrl
+
+-- | Return `true` if PyProject uses poetry build system, `false` otherwise.
+usesPoetryBuildSystem :: PyProject -> Bool
+usesPoetryBuildSystem project = case pyprojectBuildSystem project of
+  Nothing -> False
+  bs -> case (bs) of
+    Nothing -> False
+    Just bb -> (buildBackend bb) == "poetry.core.masonry.api"
+
+poetryDependencyToDependency :: [DepEnvironment] -> Text -> PoetryDependency -> Dependency
+poetryDependencyToDependency depEnvs name deps =
+  Dependency
+    { dependencyType = depType
+    , dependencyName = depName
+    , dependencyVersion = depVersion
+    , dependencyLocations = depLocations
+    , dependencyEnvironments = depEnvironment
+    , dependencyTags = depTags
+    }
+  where
+    depType = case deps of
+      (PoetryTextVersion _) -> PipType
+      (PyProjectPoetryDetailedVersionDependencySpec _) -> PipType
+      (PyProjectPoetryGitDependencySpec _) -> GitType
+      (PyProjectPoetryPathDependencySpec _) -> UserType
+      (PyProjectPoetryUrlDependencySpec _) -> URLType
+
+    depName = case deps of
+      (PoetryTextVersion _) -> name
+      (PyProjectPoetryDetailedVersionDependencySpec _) -> name
+      (PyProjectPoetryGitDependencySpec ds) -> gitUrl ds
+      (PyProjectPoetryUrlDependencySpec ds) -> sourceUrl ds
+      (PyProjectPoetryPathDependencySpec ds) -> sourcePath ds
+
+    depVersion = case deps of
+      (PoetryTextVersion ds) -> toDependencyVersion ds
+      (PyProjectPoetryDetailedVersionDependencySpec ds) -> toDependencyVersion (poetryDependencyVersion ds)
+      (PyProjectPoetryGitDependencySpec ds) -> case asum [gitTag ds, gitRev ds, gitBranch ds] of
+        Nothing -> Nothing
+        Just version -> Just $ CEq version
+      _ -> Nothing
+
+    depEnvironment = depEnvs
+    depLocations = []
+    depTags = M.empty
+
+toDependencyVersion :: Text -> Maybe VerConstraint
+toDependencyVersion dt = case parse parseConstraintExpr "" dt of
+  Left _ -> Nothing
+  Right v -> case v of
+    CEq "*" -> Nothing
+    vc -> Just vc
+
+toDependnecy :: [DepEnvironment] -> Map Text PoetryDependency -> Map Text Dependency
+toDependnecy depEnvs = M.mapWithKey $ poetryDependencyToDependency depEnvs
+
+getDependencies :: PyProject -> [Dependency]
+getDependencies project = filter ignorePython $ map snd allDeps
+  where
+    -- pyproject typically includes python as dependency that has to be ignored
+    ignorePython = (\d -> dependencyName d /= "python")
```

Comment:
[nit] I might call this `notPython`, so `filter notPython ...` reads more fluently.

Reply:
Here's an alternative implementation to wrinkle your brain a bit:

```hs
notNamedPython = (/= "python") . dependencyName
```